crates/astria-composer/src/bundler.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    ds::RollupTxExt,
+    strategy::{NoStrategy, Strategy},
+};
+use color_eyre::eyre::{self, Context};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+use tokio::sync::mpsc as tokio_mpsc;
+
+/// The Bundler module bundles received transactions based on the various
+/// MEV strategies that are implemented.
+/// It then forwards the bundles to the Executor
+pub(crate) struct Bundler {
+    collector_receiver: UnboundedReceiver<RollupTxExt>,
+    executor_sender: UnboundedSender<Vec<RollupTxExt>>,
+}
+
+impl Bundler {
+    pub(crate) fn new(
+        collector_receiver: UnboundedReceiver<RollupTxExt>,
+    ) -> (Self, UnboundedReceiver<Vec<RollupTxExt>>) {
+        let (stream, sink) = tokio_mpsc::unbounded_channel::<Vec<RollupTxExt>>();
+        let new_strategy = Self {
+            collector_receiver,
+            executor_sender: stream,
+        };
+        (new_strategy, sink)
```

Comment:
same comment as for `Collector` - i feel like we should probably only return the object that is constructed from the constructor and add a `subscribe()` method for getting a reader to its channel